Source/Core/Core/Core.cpp

```diff
@@ -351,17 +341,8 @@ static void CpuThread()
   if (_CoreParameter.bFastmem)
     EMM::InstallExceptionHandler();  // Let's run under memory watch
 
-  if (!s_state_filename.empty())
-  {
-    // Needs to PauseAndLock the Core
-    // NOTE: EmuThread should have left us in State::Stepping so nothing will happen
-    //   until after the job is serviced.
-    QueueHostJob([] {
-      // Recheck in case Movie cleared it since.
-      if (!s_state_filename.empty())
-        ::State::LoadAs(s_state_filename);
-    });
-  }
+  if (savestate_path)
+    QueueHostJob([&savestate_path] { ::State::LoadAs(*savestate_path); });
```

Comment:
IIRC this can be moved before CPU::Run() and called from this thread. So it will be deterministic and executed before the game is started at all.